### PR TITLE
Packaging with RPM

### PR DESCRIPTION
**Pull request by jhford**
This commit adds the capability to easily build an RPM for this tool.  The process to do it is to run ./build-rpm.sh.  Internally, this script prepares a tarball for the spec file, copies the files which are tracked by git into an rpmbuild SOURCES directory, builds an SRPM then uses the Fedora 25 Mock configuration to build the actual RPM.  This RPM should then be used on other systems.  I need to verify this, but we require a C++11 compiler to build the node modules, which is not currently possible.  I need to verify that the resulting RPM from a Fedora 25 build host actually runs on a CentOS 6 machine.  If it doesn't work, we could probably figure out which libraries are missing and copy those libraries into this package.  I doubt we'll need that though.  Maybe someone with a real centos 6 machine can double check this.

@djmitche I did look into building a docker image which can build this, but I found that there was a weird interaction between the mock tool and docker, around privileges needed to set up the chroot in docker.  We'd have to give the docker container a privilege called sysadmin, which sounds like a broad scope.  I suspect the solution is to build using docker without mock, but I'd rather not go down that path.

I don't know Grenade's or Arr's github handle, but they should check this out.  I'm not sure how to write the systemd or init.d script best for this tool, but it should be a matter of running /opt/taskcluster-host-secrets/bin/start.sh.  I `exec` the node process in the bash script, so it shouldn't be too hard to daemonize.

### CODE REVIEW

**Comment by jhford** [comment]
@amyrrich @grenade you should be able to build the RPM by cloning the repository on a fedora machine and running `./build-rpm.sh`.  It'll put the results into `/var/lib/mock/fedora-25-x86_64/result/taskcluster-host-secrets-1.0.0-1.fc25.x86_64.rpm`.

If you could write an init.d or systemd script to launch `/opt/taskcluster-host-secrets/bin/start.sh`, I can happily include that in the RPM.

**Comment by jhford** [comment]
this should probably use a --define to set the version to that of the package.json file...

**Comment by djmitche** [comment]
From https://github.com/mmornati/docker-mock-rpmbuilder --

    NB: It's important to run the container with privileged rights because mock needs the "unshare" system call to create a new mountpoint inside the process. Withour this you will get this error:
    ERROR: Namespace unshare failed.
    If the '--cap-add=SYS_ADMIN' is not working for you, you can run the container with the privilaged parameter. Replace '--cap-add=SYS_ADMIN' with '--privileged=true'.

That's pretty annoying, but you're right - it precludes running it in automation without adding a new docker-worker feature (probably more work than we need at the moment).

I tried installing this, but
```
[root@d7327b65a072 taskcluster-host-secrets]# ./build-rpm.sh 
+ rpmdev-setuptree
./build-rpm.sh: line 6: rpmdev-setuptree: command not found
[root@d7327b65a072 taskcluster-host-secrets]# dnf whatprovides rpmdev-setuptree
Error: No Matches found
```

what tools do I need to install in advance?

**Comment by jhford** [comment]
yah, that's the thread that I found which led me to that conclusion :/

`sudo dnf install /usr/bin/<name of command that was missing>` is a pretty good trick :)

In this case, I don't even remember the name of the package, so that's how I install it.  You'll need `/usr/bin/{rpmdev-setuptree,spectool,mock}` and I'd recommend `sudo gpasswd -a $USER mock`

File: host-secrets.spec.in [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Summary:	Taskcluster Host Secrets
+
+Autoreq: 0
+Autoprov: 0
+
+Group:		Development/Tools
+License:	MPL-2.0
+URL:		  https://github.com/taskcluster/%{name}
+Source0:	taskcluster-host-secrets.tar.gz
+Source1:  https://nodejs.org/dist/%{nodever}/node-%{nodever}-%{nodearch}.tar.xz
```
Comment (djmitche) [inline host-secrets.spec.in:23]: :heart: bundling node :)

**Review by djmitche — APPROVED** [review]

**Pull request merged**